modules/SofaConstraint/UniformConstraint.inl

```diff
@@ -51,38 +64,34 @@ void UniformConstraint<DataTypes>::getConstraintViolation(const sofa::core::Cons
     const SReal dt = this->getContext()->getDt();
     const SReal invDt = 1.0 / dt;
 
-    auto pos = this->getMState()->readPositions();
+    auto pos     = this->getMState()->readPositions();
+    auto restPos = this->getMState()->readRestPositions();
 
     if (cParams->constOrder() == sofa::core::ConstraintParams::VEL)
     {
-        const std::size_t N = Deriv::size();
-
-        for (std::size_t i = 0; i < vfree.size(); ++i)
-        {
-            for (std::size_t j = 0; j < N; ++j)
-            {
-                resV->set(m_constraintIndex + i*N + j, vfree[i][j] + invDt * pos[i][j] );
-            }
+        if (d_constraintRestPos.getValue()){
+            computeViolation(resV, m_constraintIndex, vfree, Deriv::size(),[&invDt,&pos,&vfree,&restPos](int i, int j)
```

Comment:
lambda :+1:

Reply:
Lambda is indeed nice but the indentation is strange.
I'd suggest 
```cpp
computeViolation( resV, m_constraintIndex, vfree, Deriv::size(), [&invDt, &pos, &vfree, &restPos](int i, int j) {
    return vfree[i][j] + invDt * ( pos[i][j] - restPos[i][j] );
});
```